src/Data/Auth/Providers/DatabaseAuthProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import {
+  createCookie,
+  getCookie,
+  removeCookie,
+} from "../../Common/CookieHelper";
+import { AuthContext } from "./AuthContext";
+
+interface AuthProviderProps {
+  children: React.ReactNode;
+}
+
+const DatabaseAuthProvider = ({ children }: AuthProviderProps) => {
```

Comment:
Why use the context and provider indirection? If the provider has the same interface you only construct this one and the context is the same for all of them?

Reply:
We create context in the AuthContext.ts file, which then is accessed in the application through provider, where we also create distinction in the logic between each implementation. I don't know if I explained it clearly enough.